launcher/minecraft/Rule.h

```diff
@@ -57,27 +74,43 @@ class Rule
 class OsRule : public Rule
 {
 private:
-    // the OS
-    OpSys m_system;
-    // the OS version regexp
-    QString m_version_regexp;
+    QString m_system;
 
 protected:
     virtual bool applies(const Library *)
     {
-        return (m_system == currentSystem);
+        QString sys;
+        QString arch;
+        if(m_system.contains("-"))
+        {
+            auto parts = m_system.split("-");
+            sys = parts[0];
+            arch = parts[1];
+        }
+        else
+        {
+            sys = m_system;
+        }
+        bool systemCorrect;
+        bool archCorrect = true;
+        systemCorrect = sys == SysInfo::currentSystem();
+        if(!arch.isEmpty())
+        {
+            archCorrect = arch == SysInfo::currentArch();
+        }
```

Comment:
I did some debugging and it looks like the arch comparison is never called.

Don't know why though